### PR TITLE
Wait pool in ParallelFormattingOutputFormat::finalizeImpl

### PR DESCRIPTION
**Pull request by vdimir**
### Changelog category (leave one):

- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

close #68791

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 78921fcef2bcbd66e7e97877d191a704627ecffb with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/71135/78921fcef2bcbd66e7e97877d191a704627ecffb/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/c82cf25b3e5864bcc153cbe45adb8c6527e1ec6e/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Review by novikd — APPROVED** [review]

**Comment by vdimir** [comment]
> Integration tests (asan, old analyzer) [6/6] — fail: 1, passed: 488 [Details](https://s3.amazonaws.com/clickhouse-test-reports/71135/c6c66b84e9c235e25ca5b0c6537420f590ba55e1/integration_tests__asan__old_analyzer__[6_6].html)

https://github.com/ClickHouse/ClickHouse/issues/71237

File: src/Processors/Formats/Impl/ParallelFormattingOutputFormat.cpp [hunk 1]
```diff
@@ -96,9 +97,9 @@ namespace DB
     }
 
 
-    void ParallelFormattingOutputFormat::finishAndWait() noexcept
+    void ParallelFormattingOutputFormat::finishAndWait(bool emergency_stop_) noexcept
     {
-        emergency_stop = true;
+        emergency_stop = emergency_stop_;
```
Comment (novikd) [inline src/Processors/Formats/Impl/ParallelFormattingOutputFormat.cpp:102]: What if `emergency_stop` is already set to `true`, but you set it to `false`? Let's do something like:
```
bool expected = false;
emergency_stop. compare_exchange_strong(expected, emergency_stop_);
```

**Comment by vdimir** [comment]
Test is still failing in this PR, invesitaging... https://s3.amazonaws.com/clickhouse-test-reports/71135/ad30461db871d3d8008864a566e3424a14c56769/stateless_tests__tsan__s3_storage__[2_3].html

**Comment by vdimir** [comment]
Without setting `emergency_stop` background thread continues executing, renamed it to `stop_flag`

**Comment by vdimir** [comment]
Closed in favor to https://github.com/ClickHouse/ClickHouse/pull/71304

**Pull request closed without merging**